### PR TITLE
ncm-metaconfig: cumulus: interfaces configuration

### PR DESCRIPTION
**Pull request by stdweird**

### CODE REVIEW

**Comment by kwaegema** [comment]
test this please

**Comment by jrha** [comment]
@apdibbo - you should take a look at this

**Comment by jrha** [comment]
@hpcugentbot retest this please

File: ncm-metaconfig/src/main/metaconfig/cumulus/pan/schema.pan [hunk 1]
```diff
@@ -0,0 +1,148 @@
+declaration template metaconfig/cumulus/schema;
+
+include 'pan/types';
+include 'quattor/functions/network';
+
+type cumulus_port = string; #with exists("/software/components/metaconfig/services/{/etc/network/interfaces}/ports/"
```
Comment (jrha) [inline ncm-metaconfig/src/main/metaconfig/cumulus/pan/schema.pan]: Was this intentionally left commented out?

File: ncm-metaconfig/src/main/metaconfig/cumulus/pan/schema.pan [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+
+type cumulus_interfaces = {
+    @{interfaces}
+    'interfaces' ? cumulus_interface{}
+    @{MLAG peerlink configuration}
+    'peerlink' ? cumulus_peerlink
+    @{bridge}
+    'bridge' ? cumulus_bridge
+} with {
+    # TODO
```
Comment (jrha) [inline ncm-metaconfig/src/main/metaconfig/cumulus/pan/schema.pan]: Can you open issues for what is left TODO?

**Comment by jrha** [comment]
@stdweird ping

**Comment by stdweird** [comment]
@jrha remarks addressed

File: ncm-metaconfig/src/main/metaconfig/cumulus/pan/ports.pan [hunk 3]
```diff
@@ -0,0 +1,9 @@
+unique template metaconfig/cumulus/ports;
+
+include 'metaconfig/cumulus/schema';
+
+bind "/software/components/metaconfig/services/{/etc/cumulus/ports.conf}/contents" = cumulus_ports;
+
+prefix "/software/components/metaconfig/services/{/etc/cumulus/ports.conf}";
+"module" = "cumulus/ports";
+# daemons: update-ports
```
Comment (jrha) [inline ncm-metaconfig/src/main/metaconfig/cumulus/pan/ports.pan]: Is this copypasta or useful code that got commented out?

Comment (stdweird) [reply ncm-metaconfig/src/main/metaconfig/cumulus/pan/ports.pan]: possibly useful code. i haven't run this yet on a cumulus switch
i ostly do `ifreload -a` after if copied the files manually in place

**Comment by apdibbo** [comment]
I'm just looking to see if we could use this for our config, more comments coming soon.

**Comment by apdibbo** [comment]
There are a few things which look to be missing:

- Per interface configuration of mtu - needed for using evpn and for connecting to other Vendors equipment.
- Configuration of VLAN interfaces
- Configuration of VTEP Network Interfaces (VNIs) for using VXLAN/eVPN

Here is some example config for the above
```
auto vlan10
iface vlan10
    address 192.168.1.1/24
    vlan-id 450
    vlan-raw-device pub_br

auto vni-51
iface vni-51
    bridge-access 451
    bridge-arp-nd-suppress on
    bridge-learning on
    mstpctl-bpduguard yes
    mstpctl-portbpdufilter yes
    vxlan-id 51
    mtu 9152
    vxlan-local-tunnelip 10.10.10.13

```

**Comment by apdibbo** [comment]
I also can't see anything for configuring vrf such as putting eth0 into the mgmt vrf

**Comment by jrha** [comment]
@apdibbo can you open issues for the missing functionality please?

**Comment by jrha** [comment]
@apdibbo thanks!

**Pull request merged**